File: backend/models/module_registry.py
"""
模块统一注册表 v1.0
这是整个系统的单一真相来源（Single Source of Truth）。
新增模块只需在此添加一条记录。

与前端 frontend/src/config/moduleRegistry.ts 保持同步。
"""

from dataclasses import dataclass, field
from typing import List, Dict

@dataclass
class FieldDefinition:
    """模块字段定义"""
    name: str                          # 字段名（与数据库列/JSON key 一致）
    label: str                         # 默认显示名（中文）
    type: str = "text"                 # text | number | date | select
    options: List[str] = field(default_factory=list)  # 当 type=select 时的选项
    required: bool = False
    sort_order: int = 0

@dataclass
class ModuleDefinition:
    """模块定义 - 所有元数据的唯一来源"""
    module_key: str                    # 唯一标识，如 "return_exchange"
    display_name: str                  # 默认显示名
    icon: str                          # lucide-react 图标名，如 "RotateCcw"
    route_path: str                    # 前端路由路径，如 "/return-exchange"
    navigation_group: str              # 侧边栏分组："客服业务" | "仓储业务"
    permissions: List[str]             # 权限 key，如 ["return_exchange:view"]
    fields: List[FieldDefinition]
    sort_order: int = 0
    enabled_by_default: bool = True


# ============================================================
# 注册表 - 唯一需要修改的地方
# ============================================================

MODULE_REGISTRY: Dict[str, ModuleDefinition] = {
    "tickets": ModuleDefinition(
        module_key="tickets",
        display_name="工单系统",
        icon="Ticket",
        route_path="/tickets",
        navigation_group="客服业务",
        permissions=["tickets:view"],
        sort_order=0,
        fields=[],
    ),
    "return_exchange": ModuleDefinition(
        module_key="return_exchange",
        display_name="退换登记",
        icon="RotateCcw",
        route_path="/return-exchange",
        navigation_group="客服业务",
        permissions=["return_exchange:view"],
        sort_order=1,
        fields=[
            FieldDefinition("shop_name", "店铺", sort_order=0),
            FieldDefinition("model", "型号", sort_order=1),
            FieldDefinition("config", "配置", sort_order=2),
            FieldDefinition("size", "规格", sort_order=3),
            FieldDefinition("computer_price", "电脑价格", type="number", sort_order=4),
            FieldDefinition("accessories", "配件", sort_order=5),
            FieldDefinition("accessories_price", "配件价格", type="number", sort_order=6),
            FieldDefinition("return_tracking", "寄回单号", sort_order=7),
            FieldDefinition("send_tracking", "寄出单号", sort_order=8),
            FieldDefinition("shipping_fee", "运费", type="number", sort_order=9),
            FieldDefinition("record_type", "类型", type="select",
                          options=["退货", "换货"], sort_order=10),
        ],
    ),
    "repair": ModuleDefinition(
        module_key="repair",
        display_name="维修登记",
        icon="Wrench",
        route_path="/repair",
        navigation_group="客服业务",
        permissions=["repair:view"],
        sort_order=2,
        fields=[
            FieldDefinition("model", "型号", sort_order=1),
            FieldDefinition("config", "配置", sort_order=2),
            FieldDefinition("computer_price", "电脑价格", type="number", sort_order=3),
            FieldDefinition("accessories", "配件", sort_order=4),
            FieldDefinition("return_tracking", "寄回单号", sort_order=5),
            FieldDefinition("send_tracking", "寄出单号", sort_order=6),
            FieldDefinition("shipping_fee", "运费", type="number", sort_order=7),
        ],
    ),
    "gift": ModuleDefinition(
        module_key="gift",
        display_name="发货登记",
        icon="Gift",
        route_path="/gifts",
        navigation_group="仓储业务",
        permissions=["gift:view"],
        sort_order=1,
        fields=[
            FieldDefinition("shop_name", "店铺", sort_order=0),
            FieldDefinition("model", "型号", sort_order=1),
            FieldDefinition("config", "配置", sort_order=2),
            FieldDefinition("color", "颜色", sort_order=3),
            FieldDefinition("size", "规格", sort_order=4),
            FieldDefinition("cost", "成本", type="number", sort_order=5),
            FieldDefinition("order_amount", "订单金额", type="number", sort_order=6),
            FieldDefinition("send_tracking", "寄出单号", sort_order=7),
            FieldDefinition("shipping_fee", "运费", type="number", sort_order=8),
            FieldDefinition("ship_date", "出货日期", type="date", sort_order=9),
        ],
    ),
    "gift_cashback": ModuleDefinition(
        module_key="gift_cashback",
        display_name="返现登记",
        icon="DollarSign",
        route_path="/gift-cashback",
        navigation_group="客服业务",
        permissions=["gift_cashback:view"],
        sort_order=2,
        fields=[
            FieldDefinition("cashback_amount", "返现金额", type="number", sort_order=1),
            FieldDefinition("reason", "原因", sort_order=2),
            FieldDefinition("applicant", "申请人", sort_order=3),
        ],
    ),
    "gift_resend": ModuleDefinition(
        module_key="gift_resend",
        display_name="礼品补发",
        icon="PackageCheck",
        route_path="/gift-resend",
        navigation_group="客服业务",
        permissions=["gift_resend:view"],
        sort_order=3,
        fields=[
            FieldDefinition("shop_name", "店铺", sort_order=1),
            FieldDefinition("type", "类型", sort_order=2),
            FieldDefinition("gift_detail", "礼品明细", sort_order=3),
            FieldDefinition("express_company", "快递公司", sort_order=4),
            FieldDefinition("tracking_no", "快递单号", sort_order=5),
        ],
    ),
}

File: frontend/src/config/moduleRegistry.ts
/**
 * 模块统一注册表 v1.0（前端版）
 * 与 backend/models/module_registry.py 保持同步。
 * 这个文件是前端的 Single Source of Truth。
 */

import {
  RotateCcw, Wrench, Gift, DollarSign, PackageCheck, Ticket, Laptop,
  type LucideIcon,
} from 'lucide-react'

// ============================================================
// 类型定义
// ============================================================

export interface FieldDefinition {
  name: string
  label: string
  type: 'text' | 'number' | 'date' | 'select'
  options?: string[]
  required: boolean
  sortOrder: number
}

export interface ModuleDefinition {
  moduleKey: string
  displayName: string
  icon: string           // 对应 lucide-react 图标名
  routePath: string
  navigationGroup: string
  permissions: string[]
  fields: FieldDefinition[]
  sortOrder: number
  enabledByDefault: boolean
}

// ============================================================
// 图标库 - 把字符串映射到实际组件
// ============================================================

export const ICON_MAP: Record<string, LucideIcon> = {
  RotateCcw, Wrench, Gift, DollarSign, PackageCheck, Ticket, Laptop,
}

// ============================================================
// 注册表
// ============================================================

export const MODULE_REGISTRY: Record<string, ModuleDefinition> = {
  tickets: {
    moduleKey: 'tickets',
    displayName: '工单系统',
    icon: 'Ticket',
    routePath: '/tickets',
    navigationGroup: '客服业务',
    permissions: ['tickets:view'],
    sortOrder: 0,
    enabledByDefault: true,
    fields: [],
  },
  return_exchange: {
    moduleKey: 'return_exchange',
    displayName: '退换登记',
    icon: 'RotateCcw',
    routePath: '/return-exchange',
    navigationGroup: '客服业务',
    permissions: ['return_exchange:view'],
    sortOrder: 1,
    enabledByDefault: true,
    fields: [
      { name: 'shop_name', label: '店铺', type: 'text', required: false, sortOrder: 0 },
      { name: 'model', label: '型号', type: 'text', required: false, sortOrder: 1 },
      { name: 'config', label: '配置', type: 'text', required: false, sortOrder: 2 },
      { name: 'size', label: '规格', type: 'text', required: false, sortOrder: 3 },
      { name: 'computer_price', label: '电脑价格', type: 'number', required: false, sortOrder: 4 },
      { name: 'accessories', label: '配件', type: 'text', required: false, sortOrder: 5 },
      { name: 'accessories_price', label: '配件价格', type: 'number', required: false, sortOrder: 6 },
      { name: 'return_tracking', label: '寄回单号', type: 'text', required: false, sortOrder: 7 },
      { name: 'send_tracking', label: '寄出单号', type: 'text', required: false, sortOrder: 8 },
      { name: 'shipping_fee', label: '运费', type: 'number', required: false, sortOrder: 9 },
      { name: 'record_type', label: '类型', type: 'select', options: ['退货', '换货', '升级配置'], required: false, sortOrder: 10 },
    ],
  },
  repair: {
    moduleKey: 'repair',
    displayName: '维修登记',
    icon: 'Wrench',
    routePath: '/repair',
    navigationGroup: '客服业务',
    permissions: ['repair:view'],
    sortOrder: 2,
    enabledByDefault: true,
    fields: [
      { name: 'model', label: '型号', type: 'text', required: false, sortOrder: 1 },
      { name: 'config', label: '配置', type: 'text', required: false, sortOrder: 2 },
      { name: 'computer_price', label: '电脑价格', type: 'number', required: false, sortOrder: 3 },
      { name: 'accessories', label: '配件', type: 'text', required: false, sortOrder: 4 },
      { name: 'return_tracking', label: '寄回单号', type: 'text', required: false, sortOrder: 5 },
      { name: 'send_tracking', label: '寄出单号', type: 'text', required: false, sortOrder: 6 },
      { name: 'shipping_fee', label: '运费', type: 'number', required: false, sortOrder: 7 },
    ],
  },
  gift: {
    moduleKey: 'gift',
    displayName: '发货登记',
    icon: 'Gift',
    routePath: '/gifts',
    navigationGroup: '仓储业务',
    permissions: ['gifts:view'],
    sortOrder: 1,
    enabledByDefault: true,
    fields: [
      { name: 'shop_name', label: '店铺', type: 'text', required: false, sortOrder: 0 },
      { name: 'model', label: '型号', type: 'text', required: false, sortOrder: 1 },
      { name: 'config', label: '配置', type: 'text', required: false, sortOrder: 2 },
      { name: 'color', label: '颜色', type: 'text', required: false, sortOrder: 3 },
      { name: 'size', label: '规格', type: 'text', required: false, sortOrder: 4 },
      { name: 'cost', label: '成本', type: 'number', required: false, sortOrder: 5 },
      { name: 'order_amount', label: '订单金额', type: 'number', required: false, sortOrder: 6 },
      { name: 'send_tracking', label: '寄出单号', type: 'text', required: false, sortOrder: 7 },
      { name: 'shipping_fee', label: '运费', type: 'number', required: false, sortOrder: 8 },
      { name: 'ship_date', label: '出货日期', type: 'date', required: false, sortOrder: 9 },
    ],
  },
  gift_cashback: {
    moduleKey: 'gift_cashback',
    displayName: '返现登记',
    icon: 'DollarSign',
    routePath: '/gift-cashback',
    navigationGroup: '客服业务',
    permissions: ['gift_cashback:view'],
    sortOrder: 2,
    enabledByDefault: true,
    fields: [
      { name: 'cashback_amount', label: '返现金额', type: 'number', required: false, sortOrder: 1 },
      { name: 'reason', label: '原因', type: 'text', required: false, sortOrder: 2 },
      { name: 'applicant', label: '申请人', type: 'text', required: false, sortOrder: 3 },
    ],
  },
  gift_resend: {
    moduleKey: 'gift_resend',
    displayName: '礼品补发',
    icon: 'PackageCheck',
    routePath: '/gift-resend',
    navigationGroup: '客服业务',
    permissions: ['gift_resend:view'],
    sortOrder: 3,
    enabledByDefault: true,
    fields: [
      { name: 'shop_name', label: '店铺', type: 'text', required: false, sortOrder: 1 },
      { name: 'type', label: '类型', type: 'text', required: false, sortOrder: 2 },
      { name: 'gift_detail', label: '礼品明细', type: 'text', required: false, sortOrder: 3 },
      { name: 'express_company', label: '快递公司', type: 'text', required: false, sortOrder: 4 },
      { name: 'tracking_no', label: '快递单号', type: 'text', required: false, sortOrder: 5 },
    ],
  },
  products: {
    moduleKey: 'products',
    displayName: '产品概览',
    icon: 'Laptop',
    routePath: '/products',
    navigationGroup: '主要功能',
    permissions: ['products:view'],
    sortOrder: 3,
    enabledByDefault: true,
    fields: [],
  },
}
